packages/react-devtools-extensions/src/parseHookNames/parseHookNames.js

```diff
@@ -12,28 +12,46 @@ import LRU from 'lru-cache';
 import {SourceMapConsumer} from 'source-map-js';
 import {getHookName} from '../astUtils';
 import {areSourceMapsAppliedToErrors} from '../ErrorTester';
-import {__DEBUG__} from 'react-devtools-shared/src/constants';
+import {__DEBUG__, __PROFILE__} from 'react-devtools-shared/src/constants';
 import {getHookSourceLocationKey} from 'react-devtools-shared/src/hookNamesCache';
 import {sourceMapIncludesSource} from '../SourceMapUtils';
 import {SourceMapMetadataConsumer} from '../SourceMapMetadataConsumer';
+import {
+  registerEventLogger,
+  logEvent,
+} from 'react-devtools-shared/src/EventLogger';
 
+import type {ParseHookNamesEvents} from 'react-devtools-shared/src/EventLogger';
 import type {
   HooksNode,
   HookSource,
   HooksTree,
 } from 'react-debug-tools/src/ReactDebugHooks';
 import type {HookNames, LRUCache} from 'react-devtools-shared/src/types';
-import type {Thenable} from 'shared/ReactTypes';
 import type {SourceConsumer} from '../astUtils';
 
 const MAX_SOURCE_LENGTH = 100_000_000;
 
+if (__PROFILE__) {
+  global.__loggerEvents = [];
```

Comment:
I'm wondering why we are using a global here?

Seems like we could use our feature flag system to create a proper logging (enabled for Facebook builds only).